build: Upgrade dependencies

dcalhoun opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Upgrade dependencies to the latest versions. 

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Integrate the latest bug fixes and features. 

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
* Upgrade all `@wordpress` packages. 
* Remove now unnecessary `@wordpress/components` patch. 
* Recreate existing `@wordpress` packages for new dependency versions. 
* Expand patches to export overlooked `@wordpress/editor` and `@wordpress/format-library` stylesheets. 
* Expand patches to export private modules utilized by GutenbergKit. 
* Replace usage of private `getLayoutStyles` API with new public export from the `@wordpress/global-styles-engine` package.
* Defer upgrading the following packages due to incompatibility with the Gutenberg core project:
  * `eslint` - https://github.com/WordPress/gutenberg/issues/64782
  * `react` and `react-dom` - https://github.com/WordPress/gutenberg/issues/71336
* Add `legacy-peer-deps=true` to address optional `@typescript-eslint/utils` dependency version conflict. See https://github.com/wordpress-mobile/GutenbergKit/commit/e9f99f41578965cbc057c7378dec59f908af5780. 
* Add `tsconfig.json` to address new (mistaken?) `tsconfig.json` [requirement](https://github.com/WordPress/gutenberg/commit/a4ba4519a58c63008722952feec6b6fdcf4dc75f#diff-691525462f8b4cac47c3c2e0e2668606f539ed0b3f62f8f6b79dffe7ef38bd9d) by `@wordpress/eslint-plugin`. See https://github.com/wordpress-mobile/GutenbergKit/commit/e9f99f41578965cbc057c7378dec59f908af5780. 

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
> [!tip]
> Test using the prototype builds:
> - https://github.com/wordpress-mobile/WordPress-Android/pull/22398#issuecomment-3614208868
> - https://github.com/wordpress-mobile/WordPress-iOS/pull/25047#issuecomment-3614361387

Broadly test various aspects of the editor to avoid regressions. 

### Accessibility Testing Instructions
<!-- How can you test the changes by using a keyboard/screen reader only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->
No specific user-facing changes to test. 

## Screenshots or screencast <!-- if applicable -->
No user-facing changes.

dcalhoun committed:
build: Upgrade dev dependencies to latest versions

dcalhoun committed:
build: Upgrade various `@wordpress` dependencies

dcalhoun committed:
build: Patch `@wordpress/editor` and `@wordpress/format-library` exports

Enable importing the package's CSS files, which is promoted as an option
in the Gutenberg Framework documentation: https://wordpress.org/gutenberg-framework/docs/intro#bootstrap-your-block-editor

The package `exports` field was added in: https://github.com/WordPress/gutenberg/commit/f13dcfaa60e08a4b0ac77dc105bb8b1eac055264

Without these exports, Vite's build fails with the following error:

```
(!) Failed to run dependency scan. Skipping dependency pre-bundling. Error:   Failed to scan for dependencies from entries:
  ./GutenbergKit/src/index.html

  ✘ [ERROR] Missing "./build-style/style.css" specifier in "@wordpress/format-library" package [plugin vite:dep-scan]

    src/utils/editor-styles.js:11:7:
      11 │ import '@wordpress/format-library/build-style/style.css';
         ╵        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1151:20:
      1151 │       let promise = setup({
           ╵                     ^

    at setup (file://./GutenbergKit/node_modules/vite/dist/node/chunks/config.js:31620:12)
    at handlePlugins (./GutenbergKit/node_modules/esbuild/lib/main.js:1151:21)

```

dcalhoun committed:
build: Remove `@wordpress/core-commands` usage

This existing import resulted in the following Vite build error. This
began occurring after upgrading the package. Presumably, this began
occurring due to the addition of the `exports` field to the package in:
https://github.com/WordPress/gutenberg/commit/f13dcfaa60e08a4b0ac77dc105bb8b1eac055264

```
(!) Failed to run dependency scan. Skipping dependency pre-bundling. Error: The following dependencies are imported but could not be resolved:

  @wordpress/core-commands (imported by ./GutenbergKit/src/utils/wordpress-globals.js)

Are they installed?
    at file://./GutenbergKit/node_modules/vite/dist/node/chunks/config.js:31896:33
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file://./GutenbergKit/node_modules/vite/dist/node/chunks/config.js:34104:15

```

dcalhoun committed:
build: Upgrade `@wordpress/block-library`

dcalhoun committed:
build: Upgrade `@wordpress/rich-text`

dcalhoun committed:
build: Vite `wordpress-externals-plugin` ignores raw file imports

Disable externalization of `raw` imports as the global does not provide
the required ability to inline the raw imported value.

dcalhoun committed:
build: Upgrade `@wordpress/components` to latest version

dcalhoun committed:
build: Upgrade `@wordpress/block-editor` to the latest version

dcalhoun committed:
refactor: Replace broken private `getLayoutStyles` API usage

This once private API is now a public export of the new
`@wordpress/global-styles-engine` package.

https://github.com/WordPress/gutenberg/commit/fa08e73df3#diff-3203defa3f9fa7d0f14af961ece72d1a2dabc639b1d746b975ab0d98fcfb7ab1L9-L10

dcalhoun committed:
refactor: Scope editor content styles to block canvas

Load these styles in the editor block canvas `iframe` to minimize
potential style collisions in the top-level editor. These styles are
only intended for content within the editor, not the editor UI itself.

dcalhoun committed:
fix: Improve toolbar styling consistency

Rely upon the "toolbar" `variant` configuration to ensure each toolbar
group renders a separating right border.

dcalhoun committed:
build: Fix lint errors from module resolution failures

This
`@wordpress/eslint-plugin@22.21.0` changed the import resolution
behavior. This is incompatible with the GutenbergKit project for an
unknown reason, possibly the lack of TypeScript setup.

```
Unable to resolve path to module '[module_name]'  import/no-unresolved
```

See: https://github.com/WordPress/gutenberg/blob/2d2571507d8ea76f263089c98c46110403e3eba3/packages/eslint-plugin/CHANGELOG.md#22210-2025-11-12

dcalhoun committed:
test: Avoid test failures from unlocking undefined stores

With the previous mock, the entire `@wordpress/data` module was
replaced. This led to emptying utilities like `createReduxStore`. This
resulted in other `@wordpress` modules throwing errors when attempting
to act upon stores.

```
 FAIL  src/utils/editor.test.jsx [ src/utils/editor.test.jsx ]
Error: Cannot unlock an undefined object.
 ❯ unlock node_modules/@wordpress/private-apis/src/implementation.ts:186:9
 ❯ node_modules/@wordpress/blocks/src/store/index.js:31:1
 ❯ node_modules/@wordpress/blocks/src/index.js:11:1
```

dcalhoun committed:
build: Upgrade `@vitejs/plugin-react`

The latest version of `@vitejs/plugin-react` is ESM-only. This is
incompatible with `@wordpress/eslint-plugin`'s usage of
`eslint-import-resolver-node`. Once `eslint-import-resolver-node` is
upgraded or replaced with a more capable alternative, we can remove
the `eslint-disable-line` comments.

Resulting error:

```
Unable to resolve path to module '@vitejs/plugin-react'  import/no-unresolved
```

See: https://github.com/vitejs/vite-plugin-react/issues/691#issuecomment-3172143364

dcalhoun committed:
build: Upgrade `@wordpress/eslint-plugin` to latest

`tsconfig.json` introduced as `@wordpress/eslint-plugin`'s replaced its
usage of `eslint-import-resolver-node` with
`eslint-import-resolver-typescript`. Those changes now require a
`tsconfig.json` file to ensure the resolver works properly. The
`tsconfig.json` does not add TypeScript support.

See: https://github.com/WordPress/gutenberg/commit/a4ba4519a58c63008722952feec6b6fdcf4dc75f#diff-691525462f8b4cac47c3c2e0e2668606f539ed0b3f62f8f6b79dffe7ef38bd9d

`legacy-peer-deps=true` introduced to address a dependency version
conflict. See Claude's explanation:

> 1. eslint-import-resolver-typescript@4.4.4 has an optional peer dependency on eslint-plugin-import-x
> 2. eslint-plugin-import-x requires @typescript-eslint/utils@^8.0.0
> 3. @wordpress/eslint-plugin@22.22.0 depends on @typescript-eslint/eslint-plugin@^6.4.1, which uses @typescript-eslint/utils@6.21.0
>
> So we have:
> - @typescript-eslint/utils@6.21.0 (from @wordpress/eslint-plugin chain)
> - @typescript-eslint/utils@8.48.1 (wanted by eslint-plugin-import-x, optional peer of the resolver)
>
> Since eslint-plugin-import-x is marked as optional in the resolver, using --legacy-peer-deps should be safe. The resolver will work without it - it's
> an alternative to eslint-plugin-import for those who want to use it.

Once `@wordpress/eslint-plugin` updates its `@typescript-eslint`
dependencies to v8.x, we can remove the `legacy-peer-deps` usage.

dcalhoun commented on patches/@wordpress+block-editor+15.9.0.patch:
This replaces `patches/@wordpress+block-editor+14.17.0.patch` one-for-one, aside from the inline comment below.

dcalhoun commented on patches/@wordpress+block-editor+15.9.0.patch:
```diff
@@ -0,0 +1,67 @@
+ 		"./package.json": "./package.json",
+-		"./build-style/": "./build-style/"
++		"./build-style/": "./build-style/",
++		"./build-module/components/inserter/media-tab/*": "./build-module/components/inserter/media-tab/*",
++		"./build-module/components/inserter/hooks/*": "./build-module/components/inserter/hooks/*"
```
Given GutenbergKit accesses private modules paths, we must explicitly export these modules now that `@wordpress/block-editor` [defines](https://github.com/WordPress/gutenberg/commit/525f64358cd05c64f6cc34efaa46e4d7ab0e2209) `exports` of its own.

dcalhoun commented on patches/@wordpress+block-library+9.36.0.patch:
No changes. This existing patch was recreated for the new `@wordpress/block-library` version.

dcalhoun commented on patches/@wordpress+components+29.8.0.patch:
This path was removed as it is no longer necessary after https://github.com/WordPress/gutenberg/commit/01d3959f4018cacac1e8e13358dc2642f0ee0c01.

dcalhoun commented on patches/@wordpress+editor+14.36.0.patch:
```diff
@@ -0,0 +1,14 @@
+ 		},
+-		"./package.json": "./package.json"
++		"./package.json": "./package.json",
++		"./build-style/*": "./build-style/*"
```
`@wordpress/editor` seemingly mistakenly overlooks exporting its styles. This needs to be contributed to the upstream `WordPress/gutenberg` repository.

dcalhoun commented on patches/@wordpress+format-library+5.36.0.patch:
```diff
@@ -0,0 +1,14 @@
+ 		},
+-		"./package.json": "./package.json"
++		"./package.json": "./package.json",
++		"./build-style/*": "./build-style/*"
```
`@wordpress/format-library` seemingly mistakenly overlooks exporting its styles. This needs to be contributed to the upstream `WordPress/gutenberg` repository.

dcalhoun commented on patches/@wordpress+rich-text+7.36.0.patch:
No changes. This existing patch was recreated for the new `@wordpress/rich-text` version.

dcalhoun commented on src/components/editor-toolbar/index.jsx:
```diff
@@ -118,7 +118,7 @@ const EditorToolbar = ( { className } ) => {
 				<Toolbar
 					label="Editor toolbar"
-					variant="unstyled"
+					variant="toolbar"
```
Switch to this variant to ensure separator lines between the block toolbar button groups appear consistently. 

Before | After 
-- | --
<img width="380" height="756" alt="before" src="https://github.com/user-attachments/assets/7a3bce00-1e9f-4274-8706-1501e0b594fe" /> | <img width="380" height="756" alt="after" src="https://github.com/user-attachments/assets/dad28157-1ec6-4f72-a9f0-07d5799143ae" />

kean replied:
It is way too many separators to the point they no longer serve a purpose.

I'd consider removing the separators as there is seemingly no rhyme or reason to them, and there aren't enough items in the toolbar to warrant the grouping. The important downside of adding separators as they increase the amount of horizontal space the items take.

kean replied:
Please disregard it. I open a dedicated Slack thread instead.

dcalhoun commented on src/components/visual-editor/use-editor-styles.js:
```diff
@@ -3,16 +3,9 @@
-import { unlock } from '../../lock-unlock';
-
-const { getLayoutStyles } = unlock( blockEditorPrivateApis );
+import { getLayoutStyles } from '@wordpress/global-styles-engine';
```
This is now a public API. 🎉

dcalhoun commented on src/utils/editor.test.jsx:
```diff
@@ -16,7 +16,7 @@ import { unregisterDisallowedBlocks } from './blocks';
 	store: { name: 'core/editor' },
 } ) );
-vi.mock( '@wordpress/data' );
+vi.mock( import( '@wordpress/data' ), { spy: true } );
```
Fix test failures by using the original module rather than a partial mock that caused errors within other `@wordpress` modules.

dcalhoun commented on .npmrc:
```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```
Used to fix a `@typescript-eslint/utils` dependency conflict. See https://github.com/wordpress-mobile/GutenbergKit/commit/e9f99f41578965cbc057c7378dec59f908af5780.

kean replied:
(not merge blocking)

The project doesn't currently use TypeScript and the conflict is only in its linter. Disabling the conflict detection seems like a nuclear option. It can mask more conflicts introduced in the future.

One of the alternatives:

```
  ✅ Option 2: Use npm overrides for targeted fix

  Keep the TypeScript resolver but force compatible versions:

  In package.json:
  {
  	"overrides": {
  		"eslint-import-resolver-typescript": {
  			"eslint-plugin-import-x": "$eslint-plugin-import-x",
  			"@typescript-eslint/utils": "^6.0.0"
  		}
  	}
  }

  This forces the resolver to use the v6 version of @typescript-eslint/utils that's compatible with
  @wordpress/eslint-plugin.

  Remove .npmrc: No longer needed.
```

dcalhoun replied:
I considered this, but, at the time, opted to avoid the maintenance of explicit overrides. 

I now agree with your perspective. Thanks for the note. I will update this.

dcalhoun replied:
Addressed in f36b755b4bf3d0c8b73e12b96e1bd4f419069736.

dcalhoun commented on tsconfig.json:
Does not add TypeScript support. Added to address a new (mistaken?) TypeScript configuration requirement by `@wordpress/eslint-plugin`. See https://github.com/wordpress-mobile/GutenbergKit/commit/e9f99f41578965cbc057c7378dec59f908af5780.

dcalhoun commented on vite.config.js:
```diff
@@ -66,7 +66,7 @@ function wordPressExternals() {
 				const imports = defaultImport || namedImports;
 
-				if ( module.match( /\.css\?inline$/ ) ) {
+				if ( module.match( /\.css\?(inline|raw)$/ ) ) {
```
Added for robustness and avoid cryptic. There is no active use of Vite's `raw` query parameter at the present.

dcalhoun commented on src/utils/wordpress-globals.js:
```diff
@@ -84,7 +84,6 @@ export function initializeWordPressGlobals() {
 	window.wp.commands = commands;
 	window.wp.components = components;
 	window.wp.compose = compose;
-	window.wp.coreCommands = coreCommands;
```
This module began throwing errors related to unresolved modules, likely due to the new `exports` usage in Gutenberg packages. 

I opted to remove the module entirely, as the WP Admin commands library feels unrelated to (and presumably unused in) the mobile editor.

dcalhoun committed:
fix: Remove block library theme styles

These styles overrode foundational styles causing unexpected layout
bugs. E.g., the theme styles removed the top margin from image blocks.

dcalhoun committed:
refactor: Load component styles into block editor canvas

This is no strictly necessary currently as we disable `shouldIframe`,
but it further aligns GutenbergKit with Gutenberg.

dcalhoun commented on src/components/visual-editor/index.jsx:
```diff
@@ -89,7 +91,11 @@ function VisualEditor( { hideTitle } ) {
 		// Add sensible default styles if theme styles are not present.
-		hasThemeStyles ? '' : defaultThemeStyles
+		hasThemeStyles ? '' : defaultThemeStyles,
+		componentStyles,
+		blockEditorContentStyles,
+		blocksStyles,
+		blocksEditorStyles
```
Relocated from `utils/editor-styles` to scope their loading to the block canvas `iframe`, per the [Gutenberg Framework guides](https://wordpress.org/gutenberg-framework/docs/intro#bootstrap-your-block-editor).

dcalhoun committed:
docs: Remove outdated `@wordpress/components` patch reference

The patch was removed in 6fc0ca53277cc638695e450c553ad7089fd14560.

dcalhoun commented on patches/@wordpress+components+29.8.0.patch:
This patch was removed as it is no longer necessary after https://github.com/WordPress/gutenberg/commit/01d3959f4018cacac1e8e13358dc2642f0ee0c01.

kean reviewed (approved):
Approved with two comments.
I smoke tested the changes on WoA site

dcalhoun committed:
build: Install dependencies without `legacy-peer-deps`

This was deemed to broad of a solution to a deep dependency version
conflict described in e9f99f41578965cbc057c7378dec59f908af5780.

dcalhoun committed:
build: Remove now unnecessary `gradient-parser` version override

This was added to fix a `can't find variable result` error. It appears
that since then `@wordpress/components` upgraded the version it depends
upon, making this override unnecessary.

See:

- https://github.com/wordpress-mobile/GutenbergKit/pull/70
- https://github.com/WordPress/gutenberg/pull/69783

dcalhoun committed:
build: Upgrade a few dev dependencies
